src/qml/NavigatingDrawer.qml

```diff
@@ -0,0 +1,125 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import org.qfield 1.0
+import Theme 1.0
+
+Drawer {
+  id: controller
+  width: parent.width
+  height: mainContent.height
+  Overlay.modal: null
+  modal: false
+  edge: Qt.BottomEdge
+  dragMargin: 0
+  closePolicy: Popup.NoAutoClose
+
+  property real realtimeHeight: controller.height * controller.position
+  property bool closeRequested: true
+  property bool openRequested: false
+
+  // NavigationInformationView
+  property bool navigationInformationViewEnabled: navigation.isActive
+
+  // PositioningInformationView
+  property Navigation navigation
+  property double antennaHeight
+  property bool positioningInformationViewEnabled: false
+
+  // PositioningPreciseView
+  property alias positioningPreciseView: positioningPv
+  property PositioningSettings positioningSettings
+  property Positioning positionSource
+  property bool positioningPreciseEnabled: false
+  property real positioningPreciseViewHeight
+
+  function resetHeight() {
+    let newHeight = 0
+    newHeight += (navigationInformationView.height + 8)
+    newHeight += (positioningInformationView.height + 8)
+    newHeight += (positioningPv.height + 8)
+    newHeight += 16
+    controller.height = newHeight
+  }
+
+  function updateDrawerHeight(visibleView, viewHeight){
+    if (!visibleView) {
+      controller.height -= (viewHeight + 8)
+      controller.open()
+    } else if (openRequested) {
+      controller.height += (viewHeight + 8)
+      controller.open()
+    }
+  }
+
+  onNavigationInformationViewEnabledChanged: updateDrawerHeight (navigationInformationViewEnabled, navigationInformationView.contentHeight)
+
+  onPositioningInformationViewEnabledChanged: updateDrawerHeight (positioningInformationViewEnabled, positioningInformationView.contentHeight)
+
+  onPositioningPreciseEnabledChanged: updateDrawerHeight (positioningPreciseEnabled, positioningPreciseViewHeight)
+
+  onOpenRequestedChanged: {
+    if (openRequested) {
+      closeRequested = false
+      resetHeight()
+      open()
+    } else {
+      closeRequested = true
+      close()
+    }
+  }
```

Comment:
This confuses me a bit, why do we need the complicated pair of bool property logic here? What QMl shortcoming are we fighting against?

Reply:
When the drawer is in the "Up" position, it becomes interactive and allows the user to push it down. To determine whether the user is pushing the drawer down or we are closing the drawer.

At first i was thinking we should make drawer in multiple states. `closed`, `semi open`, `expanded`. maybe user wants to collapse and see map for a short time. 

But then i removed `semi open` state to make sure every thing works as it must. 

I kept this to bring back pushed down drawer. i can make its `interactive = false` then user cannot touch it and then we can remove this lines.

Reply:
I'd rather have a cleaner set of states than a pair of booleans TBH, it'd make it easier to read/understand.

Reply:
@nirvn should i remove interaction ability? 

because right now we don't support multiple state in it. also it needs new UX definitions. for example what should we show in collapsed mode(when different elements are showing) or what should we show in expanded mode? 

if i remove the interaction this complexity will be removed.

Reply:
@mohsenD98 , yes, let's remove for now.